molgenis-app/src/test/java/org/molgenis/app/promise/PromiseClientImplTest.java

```diff
@@ -0,0 +1,60 @@
+package org.molgenis.app.promise;
+
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+
+@ContextConfiguration
+public class PromiseClientImplTest extends AbstractTestNGSpringContextTests
+{
+	// @Configuration
+	// static class Config
+	// {
+	// @Bean
+	// public PromiseClient promiseClient()
+	// {
+	// return new PromiseClientImpl(soapConnectionFactory());
+	// }
+	//
+	// @Bean
+	// public SOAPConnectionFactory soapConnectionFactory()
```

Comment:
http://docs.spring.io/spring-ws/site/reference/html/client.html
WebServiceTemplate would probably allow you to stub this quite effectively (and lose some of the low-level soap code)